src/EntityFramework.SqlServer/SqlProviderServices.cs

```diff
@@ -132,6 +133,16 @@ public static bool TruncateDecimalsToScale
         }
 
         /// <summary>
+        /// Set this flag to false to prevent ordering using a row_number() function for queries with OFFSET operation, and order by columns directly instead.
+        /// </summary>
+        /// <remarks>Using row_number() can reduce ambiguity of the ordering result for some cases.</remarks>
+        public static bool UseOrderByRowNumberForQueriesWithOffset
```

Comment:
UseLegacyOrderingForQueriesWithOffset? not sure if I like it better tho, any other suggestions for name?